pkg/api/meta/priority.go

```diff
@@ -153,7 +153,59 @@ func kindMatches(pattern unversioned.GroupVersionKind, kind unversioned.GroupVer
 }
 
 func (m PriorityRESTMapper) RESTMapping(gk unversioned.GroupKind, versions ...string) (mapping *RESTMapping, err error) {
-	return m.Delegate.RESTMapping(gk, versions...)
+	mappings, err := m.Delegate.RESTMappings(gk)
+	if err != nil {
+		return nil, err
+	}
+
+	// any versions the user provides take priority
+	var priorities []unversioned.GroupVersionKind
+	if len(versions) > 0 {
+		priorities = make([]unversioned.GroupVersionKind, 0, len(m.KindPriority)+len(versions))
```

Comment:
actually you could even remove the else and set the inital value of priorities to m.KindPriority like:

```go
priorities := m.KindPriority
for _, version := range versions {
    gv, err := unversioned.ParseGroupVersion(version)
    if err != nil {
        return nil, err
    }
    priorities = append(priorities, gv.WithKind(AnyKind))
}
}
```